orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -75,6 +76,30 @@ void ResolveExtraProviderOptions(const std::vector<std::string>& provider_types,
   }
 }
 
+std::vector<std::shared_ptr<IExecutionProvider>> GetExecutionProviders(int device_type, int device_id) {
+  std::vector<std::shared_ptr<IExecutionProvider>> provider;
+
+#ifdef USE_CUDA
+  if (device_type == 1) {
+    OrtCUDAProviderOptions provider_options{};
+    provider_options.do_copy_in_default_stream = true;
+    if (device_id) {
+      provider_options.device_id = device_id;
+    }
+    if (auto factory = CudaProviderFactoryCreator::Create(&provider_options))
+      provider.push_back(factory->CreateProvider());
+  }
+#endif
+  ORT_UNUSED_PARAMETER(device_id);
+  if (device_type == 0) {
+    provider = std::vector<std::shared_ptr<IExecutionProvider>>();
+  } else {
+    ORT_THROW("Unsupported device type: ", device_type);
+  }
```

Comment:
This logic does not seem right. If the `device_type` == 1, then it will always throw.
You're missing a return statement. You should also define `ORT_UNUSED_PARAMETER` only when `USE_CUDA` is not defined.

Reply:
what do you mean by it will always throw? if theres no USE_CUDA, it will skip that part, if its not built using cuda, we don't expect user to pass cuda as a type.

Reply:
let's say `USE_CUDA` is true. And user provided device_type as 1. Then the code will hit the `USE_CUDA` block execute that and proceed to the remainder of the code. And then it sees that `device_type` != 0, will go into the else block and throw an error.